x-pack/plugins/stack_connectors/server/connector_types/resilient/resilient.ts

```diff
@@ -0,0 +1,363 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AxiosError } from 'axios';
+import { omitBy, isNil } from 'lodash/fp';
+import { CaseConnector, ServiceParams } from '@kbn/actions-plugin/server';
+import { schema } from '@kbn/config-schema';
+import { getErrorMessage } from '@kbn/actions-plugin/server/lib/axios_utils';
+import {
+  CreateIncidentData,
+  ExternalServiceIncidentResponse,
+  GetIncidentTypesResponse,
+  GetSeverityResponse,
+  Incident,
+  ResilientConfig,
+  ResilientSecrets,
+  UpdateIncidentParams,
+} from './types';
+import * as i18n from './translations';
+import { SUB_ACTION } from './constants';
+import {
+  ExecutorSubActionCommonFieldsParamsSchema,
+  ExecutorSubActionGetIncidentParamsSchema,
+  ExecutorSubActionGetIncidentTypesParamsSchema,
+  ExecutorSubActionGetSeverityParamsSchema,
+  ExecutorSubActionHandshakeParamsSchema,
+  GetCommonFieldsResponseSchema,
+  GetIncidentTypesResponseSchema,
+  GetSeverityResponseSchema,
+} from './schema';
+import { formatUpdateRequest } from './utils';
+
+const VIEW_INCIDENT_URL = `#incidents`;
+
+export class ResilientConnector extends CaseConnector<ResilientConfig, ResilientSecrets> {
+  private urls: {
+    incidentTypes: string;
+    incident: string;
+    comment: string;
+    severity: string;
+  };
+
+  constructor(params: ServiceParams<ResilientConfig, ResilientSecrets>) {
+    super(params);
+
+    this.urls = {
+      incidentTypes: `${this.getIncidentFieldsUrl()}/incident_type_ids`,
+      incident: `${this.getOrgUrl()}/incidents`,
+      comment: `${this.getOrgUrl()}/incidents/{inc_id}/comments`,
+      severity: `${this.getIncidentFieldsUrl()}/severity_code`,
+    };
+
+    this.registerSubActions();
+  }
+
+  protected getResponseErrorMessage(error: AxiosError) {
+    if (!error.response?.status) {
+      return i18n.UNKNOWN_API_ERROR;
+    }
+    if (error.response.status === 401) {
```

Comment:
To separate `unauthorized` error from the rest.

Reply:
Yes but what makes it more important than the others? It seems that this status code get its dedicated translation but the others not.

Reply:
Because I thought `Unauthrozied` is a specific error and should be logged as `Unauthrozied` rather than unknown APIerror. But I am fine with providing one generic error as well.

Reply:
No nothing specific about it. I will keep 401 as separate error